openbb_platform/CONTRIBUTING.md

```diff
@@ -186,248 +124,248 @@ class StockEODData(Data):
 
-## Using other extension as a dependency
+> Note that sometimes manual intervention is needed. For example, adjusting out-of-top level imports or adding specific arguments for a given fetcher.
 
-We can use internal and external extensions with the custom developed extension and bundle it as a dependency.
+### Integration tests
 
-### Using our internal extension
+The integration tests are a bit more complex than the unit tests, as we want to test both the Python interface and the API interface. For this, we have two scripts that will help you generate the integration tests.
 
-We will use the `openbb-qa` extension by utilizing its `summary` endpoint.
+To generate the integration tests for the Python interface, you can run the following command:
 
-To create a `period_summary_example` endpoint we need to add the following to the `router.py` file:
+```bash
+python openbb_platform/extensions/tests/utils/integration_tests_generator.py
+```
 
-```python
-from typing import List
-from openbb_provider.abstract.data import Data
-from openbb_core.app.utils import basemodel_to_df, df_to_basemodel
-from openbb_qa.qa_router import summary
-
-
-@router.command(methods=["POST"])
-def period_summary_example(
-    data: List[Data],
-    target: str,
-    start_date: str,
-    end_date: str,
-) -> OBBject[dict]:
-    """Example Data."""
+To generate the integration tests for the API interface, you can run the following command:
 
-    df = basemodel_to_df(data)
-    df = df[(df["date"] >= start_date) & (df["date"] <= end_date)]
-    df = df.reset_index(drop=True)
-    target_data = df_to_basemodel(df)
-    results = summary(target_data, target=target)
+```bash
+python openbb_platform/extensions/tests/utils/integration_tests_api_generator.py
+```
+
+When testing the API interface, you'll need to run the OpenBB Platform locally before running the tests. To do so, you can run the following command:
 
-    return OBBject(results=results.results)
+```bash
+uvicorn openbb_platform.platform.core.openbb_core.api.rest_api:app --host 0.0.0.0 --port 8000 --reload
 ```
 
-Here we are using two vital utility functions - `basemodel_to_df` and `df_to_basemodel`. The function `basemodel_to_df` converts the data to a pandas dataframe and `df_to_basemodel` converts the dataframe back to the `Data` model.
+These automated tests are a great way to reduce the amount of code you need to write, but they are not a replacement for manual testing and might require tweaking. That's why we have unit tests that test the generated integration tests to ensure they cover all providers and parameters.
 
-### Adding an external extension
+To run the tests we can do:
 
-To add the `openbb-charting` charting extension as a dependency, you'll need to add it to the `pyproject.toml` file:
+```bash
+pytest openbb_platform -m "not integration" --> only unit tests
+pytest openbb_platform -m "integration" --> only integration tests
+pytest openbb_platform --> everything
+```
```

Comment:
```suggestion
Unit tests only:

```bash
pytest openbb_platform -m "not integration"
```

Integration tests only:

```bash
pytest openbb_platform -m "integration"
```

Both unit and Integration tests:

```bash
pytest openbb_platform --> everything
```
```